app/renderer/components/tabs/content/favIcon.js

```diff
@@ -1,86 +1,87 @@
+    props.tabLoading = faviconState.showLoadingIcon(currentWindow, frameKey)
     props.tabIconColor = tabUIState.getTabIconColor(currentWindow, frameKey)
-    props.isNarrowestView = tabUIState.isNarrowestView(currentWindow, frameKey)
-    props.showFavIcon = !((hasBreakpoint(ownProps.breakpoint, 'extraSmall') && ownProps.isActive) || frame.get('location') === 'about:newtab')
-    // used in functions
-    props.frameKey = frameKey
+    props.showIconWithLessMargin = faviconState.showIconWithLessMargin(currentWindow, frameKey)
+    props.showIconAtReducedSize = faviconState.showFaviconAtReducedSize(currentWindow, frameKey)
+    props.tabId = tabId
 
     return props
   }
 
+  get testingIcon () {
+    // this is only for testing purposes
+    return this.props.tabLoading
+      ? 'loading'
+      : this.props.favicon || 'defaultIcon'
+  }
+
   render () {
-    if (!this.props.showFavIcon) {
+    if (!this.props.isPinned && !this.props.showIcon) {
       return null
     }
+
     const iconStyles = StyleSheet.create({
-      favicon: {
+      icon__loading_color: {
+        filter: this.props.tabIconColor === 'white' ? filter.makeWhite : 'none'
+      },
+      icon__favicon: {
         backgroundImage: `url(${this.props.favicon})`,
-        filter: this.props.tabIconColor === 'white' ? globalStyles.filter.whiteShadow : 'none'
+        filter: this.props.tabIconColor === 'white' ? filter.whiteShadow : 'none'
       },
-      loadingIconColor: {
-        // Don't change icon color unless when it should be white
-        filter: this.props.tabIconColor === 'white' ? globalStyles.filter.makeWhite : 'none'
+      icon__default_props: {
+        WebkitMaskSize: this.props.showIconAtReducedSize ? '10px' : '12px',
+        backgroundColor: this.props.tabIconColor === 'white' ? color.white100 : color.mediumGray
       }
     })
 
     return <TabIcon
-      data-test-favicon={this.props.favicon}
-      data-test-id={this.props.deprecatedIsTabLoading ? 'loading' : 'defaultIcon'}
+      data-test-id={this.testingIcon}
       className={css(
         styles.icon,
-        this.props.favicon && iconStyles.favicon,
-        !this.props.isPinnedTab && this.props.isNarrowestView && styles.faviconNarrowView
+        this.props.favicon && iconStyles.icon__favicon,
+        !this.props.isPinned && this.props.showIconWithLessMargin && styles.icon_lessMargin,
+        !this.props.isPinned && this.props.showIconAtReducedSize && styles.icon_reducedSize
       )}
       symbol={
-        (this.props.deprecatedIsTabLoading && css(styles.loadingIcon, iconStyles.loadingIconColor)) ||
-        this.defaultIcon
+        this.props.tabLoading
+          ? (
+            // no loading icon if there's no room for the icon
+            !this.props.showIconAtReducedSize &&
+            css(styles.icon__loading, iconStyles.icon__loading_color)
+          )
+          : !this.props.favicon && css(styles.icon__default, iconStyles.icon__default_props)
```

Comment:
We have `icon__default*` for both styles and iconStyles. I'm wondering if it would be confusing a bit.

Reply:
changed `icon__default_props` to `icon__default_sizeAndColor`. Not sure if best but more declarative ya

Reply:
Confirmed this was addressed on https://github.com/brave/browser-laptop/pull/10691/files#diff-af59f73d2e4394ed5cd93fbab7e1cdc7R92

On the commit: https://github.com/brave/browser-laptop/commit/c6947c52b8bbb073cdd257e3e243418e7b4544de#diff-af59f73d2e4394ed5cd93fbab7e1cdc7R92